compiler/damlc/tests/src/DA/Test/DamlcMultiPackage.hs

```diff
@@ -0,0 +1,760 @@
+            (["--all", "--no-cache"], "") -- Cache disabled
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- So second time rebuilds everything
+            simpleTwoPackageProject
+        , testCache
+            "Just B rebuilds if its code is modified"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            (const $ appendFile "./package-b/daml/PackageBMain.daml" "\nmyDef = 3") -- Modify package-b/daml/PackageBMain.daml
+            (["--all"], "")
+            [PackageIdentifier "package-b" "0.0.1"] -- So second time rebuilds only B, as nothing depends on B
+            simpleTwoPackageProject
+        , testCache
+            "A and B rebuild if A's code is modified"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            (const $ appendFile "./package-a/daml/PackageAMain.daml" "\nmyDef = 3") -- Modify package-a/daml/PackageAMain.daml
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- So second time rebuilds A and B, as B depends on A
+            simpleTwoPackageProject
+        , testCache
+            "Only A is rebuild if its Dar is deleted but it's package-id doesn't change"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            (const $ removeFile "package-a/.daml/dist/package-a-0.0.1.dar") -- Delete package-a/.daml/dist/package-a-0.0.1.dar
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1"] -- So second time rebuilds only A, as its package-id hasn't changed so B is not stale
+            simpleTwoPackageProject
+        , testCache
+            "A and B rebuild if A's code is modified and A's dar is deleted"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            -- Modify package-a/daml/PackageAMain.daml, Delete package-a/.daml/dist/package-a-0.0.1.dar
+            (const $ do
+              appendFile "./package-a/daml/PackageAMain.daml" "\nmyDef = 3"
+              removeFile "package-a/.daml/dist/package-a-0.0.1.dar"
+            )
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- So second time rebuilds A and B, as B depends on A
+            simpleTwoPackageProject
+        , testCache
+            "B rebuilds is A is manually rebuilt after change"
+            (["--all"], "")
+            [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"] -- First time builds both
+            -- Modify package-a/daml/PackageAMain.daml, Delete package-a/.daml/dist/package-a-0.0.1.dar
```

Comment:
```suggestion
            -- Modify package-a/daml/PackageAMain.daml, manually rebuild package-a/.daml/dist/package-a-0.0.1.dar
```

Reply:
for completeness i'd also like to see a test where we delete `a.dar`, manually rebuild it without changing anything else, then `multibuild --all`; iiuc nothing should get rebuilt

Reply:
I can add this case, but I'll note that manually building always generates a dar, and overrides regardless of if one already existed, so this would be "identical" to the above case

Reply:
oh, I thought multi-package would figure out that it doesn't need to rebuild `b` if `a`'s package-id remains unchanged, as in the case `"Only A is rebuild if its Dar is deleted but it's package-id doesn't change"`, even if the `a` dar was built manually

Reply:
> iiuc nothing should get rebuilt

this should have been: "iiuc nothing should get rebuilt _in the second multi-package invocation_"

Reply:
Yes yes, I'm just saying that manually rebuilding a DAR, vs deleting it then manually rebuilding it, is the same thing, as manually rebuild overwrites the dar.
And as such, the resulting multi-package behaviour for those cases is the same too.